tests/unit/scheduler.c

```diff
@@ -91,12 +91,14 @@ int test_multiplexing_schedule()
 	ret = create_multiplexed(size, num_rails, align, &schedule);
 	if (ret) {
 		NCCL_OFI_WARN("Failed to create multiplexed schedule");
+		free(ref_schedule);
```

Comment:
This seems like the type of thing one might use a `goto` for, so you only have to `free(ref_schedule)` once.

Reply:
Let's do this.

Reply:
Decided against this, the rest of this unit test with expanded context is a series of individual tests in the middle of a single function, using gotos would make it much harder to read. We would benefit from a unit test framework.